shell/platform/android/io/flutter/embedding/engine/FlutterJNI.java

```diff
@@ -93,18 +94,73 @@
 public class FlutterJNI {
   private static final String TAG = "FlutterJNI";
 
+  @Nullable
+  private static AsyncWaitForVsyncDelegate asyncWaitForVsyncDelegate;
+  @Nullable
+  private static Long lastVsyncCookie;
+  // This should also be updated by FlutterView when it is attached to a Display.
+  // The initial value of 0.0 indicates unknown refresh rate.
+  private static float refreshRateFPS = 0.0f;
+
   // This is set from native code via JNI.
   @Nullable
   private static String observatoryUri;
 
+  // TODO(mattcarroll): add javadocs
+  public static native void nativeInit(
+      @NonNull Context context,
+      @NonNull String[] args,
+      @Nullable String bundlePath,
+      @NonNull String appStoragePath,
+      @NonNull String engineCachesPath
+  );
+
+  // TODO(mattcarroll): add javadocs
+  public static native void nativeRecordStartTimestamp(long initTimeMillis);
+
+  // TODO(mattcarroll): add javadocs
   @UiThread
   public static native boolean nativeGetIsSoftwareRenderingEnabled();
 
   @Nullable
+  // TODO(mattcarroll): add javadocs
   public static String getObservatoryUri() {
     return observatoryUri;
   }
 
+  public static void setRefreshRateFPS(float refreshRateFPS) {
+    FlutterJNI.refreshRateFPS = refreshRateFPS;
+  }
+
+  // TODO(mattcarroll): add javadocs
+  public static void setAsyncWaitForVsyncDelegate(@Nullable AsyncWaitForVsyncDelegate delegate) {
```

Comment:
Can the `lastVsyncCookie` mechanism be moved into the VsyncWaiter?

I'd like to keep FlutterJNI as minimal as possible and have the `asyncWaitForVsync` function called from native just jump into the real implementation in VsyncWaiter.